Ianvs v0.1 code and document

jaypume opened the pull request.

Commit:
Initial commit

Commit:
add the ability to parse config file

Signed-off-by: JimmyYang20 <yangjin39@huawei.com>

Commit:
support local benchmarking

Signed-off-by: JimmyYang20 <yangjin39@huawei.com>

MooreZheng committed:
proposal of ianvs

MooreZheng committed:
images for ianvs proposal

MooreZheng committed:
Update ianvs.md

MooreZheng committed:
Update ianvs.md

MooreZheng committed:
Update ianvs.md

fixed typo

MooreZheng committed:
Update ianvs.md

update goals

MooreZheng committed:
Update ianvs.md

update sections

MooreZheng committed:
Update ianvs.md

polish

MooreZheng committed:
Update ianvs.md

update proposal

MooreZheng committed:
Update ianvs.md

MooreZheng committed:
Update ianvs.md

polish intro

MooreZheng committed:
Rename ianvs.md to Distributed Synergy AI Benchmarking.md

MooreZheng committed:
Update Distributed Synergy AI Benchmarking.md

MooreZheng committed:
Update Distributed Synergy AI Benchmarking.md

MooreZheng committed:
Add events not included in ianvs

MooreZheng committed:
Polish steps and components of ianvs

MooreZheng committed:
Modify as adviced on KubeEdge routine meeting

MooreZheng committed:
Add roadmap and adjust design details

MooreZheng committed:
Update Distributed Synergy AI Benchmarking.md

MooreZheng committed:
Modify first step and the module naming

MooreZheng committed:
Fixed background color

MooreZheng committed:
Update Distributed Synergy AI Benchmarking.md

fixed typo

MooreZheng committed:
directory restructure

MooreZheng committed:
Update distributed-synergy-ai-benchmarking.md

fixed link

MooreZheng committed:
write index.rst and fix file naming

MooreZheng committed:
Rename basicil-fpn.md to basicIL-fpn.md

MooreZheng committed:
update quick-start

MooreZheng committed:
update roadmap

Signed-off-by: Zimu Zheng <zimu.zheng@hotmail.com>

MooreZheng committed:
update how-to-test-algorithms.md

Signed-off-by: Zimu Zheng <zimu.zheng@hotmail.com>

MooreZheng committed:
update title

MooreZheng committed:
for website

MooreZheng committed:
update for website

MooreZheng committed:
update conf.py for website

MooreZheng committed:
update conf

MooreZheng committed:
update conf

Signed-off-by: Zimu Zheng <zimu.zheng@hotmail.com>

MooreZheng committed:
update conf

MooreZheng committed:
update conf

MooreZheng committed:
update conf

MooreZheng committed:
update conf

MooreZheng committed:
update for website index

jaypume committed:
Add Leaderboard of single task learning

jaypume committed:
Add contribute-leaderboards-or-test-reports

MooreZheng committed:
update how to test

jaypume committed:
Add contribute-test-environments

MooreZheng committed:
update pcb-aoi dataset

MooreZheng committed:
Merge pull request #1 from jaypume/doc

Doc update

MooreZheng committed:
Update index.rst

MooreZheng committed:
Update how-to-test-algorithms.md

MooreZheng committed:
Update index.rst

MooreZheng committed:
Update index.rst

MooreZheng committed:
Update index.rst

MooreZheng committed:
Update index.rst

MooreZheng committed:
update dataset description

MooreZheng committed:
update dataset description

MooreZheng committed:
add test reports

jaypume committed:
Add FPN & BasicIL algorithm

Commit:
update github workflow

Signed-off-by: JimmyYang20 <yangjin39@huawei.com>

Commit:
update

Signed-off-by: JimmyYang20 <yangjin39@huawei.com>

Commit:
update

Signed-off-by: JimmyYang20 <yangjin39@huawei.com>

Commit:
update

Signed-off-by: JimmyYang20 <yangjin39@huawei.com>

Commit:
update

Signed-off-by: JimmyYang20 <yangjin39@huawei.com>

jaypume committed:
update basicIL-fpn, fpn

jaypume committed:
Add section: how-to-contribute-algorithms

jaypume committed:
Add section: how-to-install-ianvs

MooreZheng committed:
Merge pull request #2 from jaypume/doc

Doc update

jaypume committed:
minor fix to section: how-to-install-ianvs

MooreZheng committed:
Merge pull request #3 from jaypume/doc

minor fix to section: how-to-install-ianvs

Commit:
update

Signed-off-by: JimmyYang20 <yangjin39@huawei.com>

MooreZheng committed:
update quick start and testing algorithm

Signed-off-by: Zimu Zheng <zimu.zheng@hotmail.com>

MooreZheng committed:
update test reports

Signed-off-by: Zimu Zheng <zimu.zheng@hotmail.com>

MooreZheng committed:
Update quick-start.md

highlight shell

Commit:
update

Signed-off-by: JimmyYang20 <yangjin39@huawei.com>

MooreZheng committed:
Update distributed-synergy-ai-benchmarking.md

Commit:
update

Signed-off-by: JimmyYang20 <yangjin39@huawei.com>

MooreZheng committed:
update quick-start

Commit:
update

Signed-off-by: JimmyYang20 <yangjin39@huawei.com>

jaypume committed:
Add section: leaderboard-of-incremental-learning

MooreZheng committed:
Merge pull request #4 from jaypume/doc

Add section: leaderboard-of-incremental-learning

MooreZheng committed:
Merge branch 'main' of https://github.com/MooreZheng/ianvs

MooreZheng committed:
Update quickstart

Signed-off-by: Zimu Zheng <zimu.zheng@hotmail.com>

Commit:
Merge pull request #4 from MooreZheng/main

Add documentation

back1860 committed:
delete the url in single task learning leaderboard and incremental learning leaderboard

MooreZheng committed:
Update README.md and related documents

Signed-off-by: Zimu Zheng <zimu.zheng@hotmail.com>

back1860 committed:
Add FPN introduction

MooreZheng committed:
Update meeting time as biweekly

Signed-off-by: Zimu Zheng <zimu.zheng@hotmail.com>

MooreZheng committed:
Merge pull request #1 from back1860/main

delete the url in single task learning leaderboard and incremental le…

MooreZheng committed:
Merge pull request #2 from MooreZheng/main

Update README.md and related documents

jaypume committed:
Update section: how-to-contribute-test-environments

jaypume committed:
Update section: how-to-install-ianvs

jaypume committed:
Update section: how-to-install-ianvs

jaypume committed:
Add conda environment creating steps

jaypume committed:
Update: how-to-install-ianvs

jaypume commented on .vscode/settings.json:
```diff
@@ -0,0 +1,3 @@
+{
```
Your local configuration file should not commit to the public repository.
Please remove this file and directory.

MooreZheng replied:
Thank you for comment. Revised as suggested and add .gitignore accordinly. See commit: https://github.com/kubeedge-sedna/ianvs/pull/4/commits/1d79bb939aba1594b785e636d89dd37591cb68c6

jaypume commented on core/__init__.py:
```diff
@@ -0,0 +1,15 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from .__version__ import __version__
```
No newline at the end of file, please add an empty line.
This also needs to be considered in the remaining python files.

MooreZheng replied:
Thank you for comment. Revised as suggested and add .gitignore
https://github.com/kubeedge-sedna/ianvs/pull/4/commits/1d79bb939aba1594b785e636d89dd37591cb68c6

MooreZheng committed:
Add scope to the homepage

Signed-off-by: Zimu Zheng <zimu.zheng@hotmail.com>

MooreZheng committed:
Merge pull request #5 from MooreZheng/v0.1

Add scope to the homepage

MooreZheng commented on docs/proposals/algorithms/single-task-learning/fpn.md:
```diff
@@ -0,0 +1,177 @@
+# Single task learning: FPN
+
+It is a traditional learning pooling all data together to train a single model. It typically includes a specialist model laser-focused on a single task and requires large amounts of task-specific labeled data, which is not always available on early stage of a distributed synergy AI project. 
```
The word "It" in the first sentence may not be clear to readers. Maybe it would be better to directly refer "single task learning" here.

MooreZheng replied:
Polished. See commit: https://github.com/kubeedge-sedna/ianvs/pull/4/commits/0bf938cf1f25680b6caa457ba2fbf6df3b7b1ec6

jaypume committed:
Merge pull request #3 from jaypume/doc

Doc update

MooreZheng committed:
Re-organize concepts and modify images accordingly

Signed-off-by: Zimu Zheng <zimu.zheng@hotmail.com>

MooreZheng committed:
Merge pull request #6 from MooreZheng/v0.1

Re-organize concepts and modify images accordingly

MooreZheng committed:
Remove ./vscode related issue

Signed-off-by: Zimu Zheng <zimu.zheng@hotmail.com>

MooreZheng committed:
Remove ./vscode related doc - Merge pull request #7 from MooreZheng/v0.1

Remove ./vscode related doc

MooreZheng committed:
Add scope of simulation tools

Signed-off-by: Zimu Zheng <zimu.zheng@hotmail.com>

MooreZheng committed:
Add scope of simulation tools - Merge pull request #8 from MooreZheng/v0.1

Add scope of simulation tools

Commit:
optimize code to reduce the delay

Signed-off-by: JimmyYang20 <yangjin39@huawei.com>

MooreZheng commented on docs/guides/how-to-contribute-leaderboards-or-test-reports.md:
```diff
@@ -0,0 +1,54 @@
+If you follow this guide and find some problem, please fill an issue to update this file.
+
+## 1. Install Tools
+### Install Git
```
It looks strange to have three preparation steps but only one submit step. A little bit more details on how to write a test report is also suggested.

MooreZheng commented on docs/guides/how-to-contribute-leaderboards-or-test-reports.md:
```diff
@@ -0,0 +1,54 @@
+# How to contribute leaderboards or test reports
+
+
+This document helps you prepare environment for developing code for Ianvs.
```
Not environment though. This paragraph serves as the very beginning which need to polish carefully.

MooreZheng replied:
Have revised in commit https://github.com/kubeedge-sedna/ianvs/pull/4/commits/9f4ba479e914ab9847adfc80617d2e0b6ead77e5

MooreZheng commented on docs/guides/how-to-contribute-leaderboards-or-test-reports.md:
```diff
@@ -0,0 +1,54 @@
+# How to contribute leaderboards or test reports
```
Leaderboard is not for every one to commit. Rules needed.

MooreZheng replied:
Fixed in commit https://github.com/kubeedge-sedna/ianvs/pull/4/commits/9f4ba479e914ab9847adfc80617d2e0b6ead77e5

MooreZheng commented on docs/guides/how-to-contribute-leaderboards-or-test-reports.md:
```diff
@@ -0,0 +1,54 @@
+
+## 2. Clone the code
+
+Clone the `Ianvs` repo:
```
Should not be `folk` instead of `clone`?

Commit:
add hard example mining algorithm test

Signed-off-by: JimmyYang20 <yangjin39@huawei.com>

MooreZheng committed:
Add leaderboard contribute solution

Signed-off-by: Zimu Zheng <zimu.zheng@hotmail.com>

MooreZheng committed:
Merge branch 'kubeedge-sedna:v0.1' into v0.1

MooreZheng committed:
Merge pull request #9 from MooreZheng/v0.1

Add leaderboard contribute solution

MooreZheng committed:
Add data source and model source link; Update data explorer

Signed-off-by: Zimu Zheng <zimu.zheng@hotmail.com>

MooreZheng committed:
Merge branch 'kubeedge-sedna:v0.1' into v0.1

MooreZheng committed:
Merge pull request #10 from MooreZheng/v0.1

Add data source and model source link; Update data explorer

MooreZheng committed:
fix figures

Signed-off-by: Zimu Zheng <zimu.zheng@hotmail.com>

MooreZheng committed:
link quick-start to YAMLs

Signed-off-by: Zimu Zheng <zimu.zheng@hotmail.com>

MooreZheng committed:
Merge pull request #11 from MooreZheng/v0.1

link quick-start to YAMLs

MooreZheng committed:
add algorithm procedures

Signed-off-by: Zimu Zheng <zimu.zheng@hotmail.com>

MooreZheng committed:
Merge pull request #12 from MooreZheng/v0.1

add algorithm procedures

Commit:
Add new metric: data transfer count ratio

Signed-off-by: JimmyYang20 <yangjin39@huawei.com>

Commit:
Add main command-line: -v

Signed-off-by: JimmyYang20 <yangjin39@huawei.com>

Commit:
Optimize the code based on pylint detection

1. add code comment.
2. clean code

Signed-off-by: JimmyYang20 <yangjin39@huawei.com>

Commit:
Fix pylint of ci

Signed-off-by: JimmyYang20 <yangjin39@huawei.com>

Commit:
Add codeql

Signed-off-by: JimmyYang20 <yangjin39@huawei.com>

Commit:
Fix licneses of third packages

Signed-off-by: JimmyYang20 <yangjin39@huawei.com>

Commit:
Add ci and license scan reports

Signed-off-by: JimmyYang20 <yangjin39@huawei.com>

Commit:
Fix hyperlink of CI

Signed-off-by: JimmyYang20 <yangjin39@huawei.com>

Commit:
Supplementary comments for closing the global warning log

Signed-off-by: JimmyYang20 <yangjin39@huawei.com>

jaypume commented on examples/pcb-aoi/incremental_learning_bench/testalgorithms/fpn/fpn_algorithm.yaml:
```diff
@@ -0,0 +1,56 @@
+            values:
+              - "./examples/pcb-aoi/incremental_learning_bench/testalgorithms/fpn/fpn_hyperparameter.yaml"
+      #  2> "hard_example_mining": check hard example when predict ; optional module;
+    - kind: "hard_example_mining"
```
It is suggested to explain why there are two modules, and also it is needed to provide a unified module definition. For example, the two module can be corresponding to the module in Sedna Incremental Learning. At least, it needs to be added about the description of how they now defined

Commit:
Convert kind to type

Signed-off-by: JimmyYang20 <yangjin39@huawei.com>

jaypume committed:
Move leaderboard outside proposal

Signed-off-by: Jie Pu <pujie2@huawei.com>

MooreZheng commented on examples/pcb-aoi/singletask_learning_bench/benchmarkingjob.yaml:
```diff
@@ -0,0 +1,67 @@
+  testenv: "./examples/pcb-aoi/singletask_learning_bench/testenv/testenv.yaml"
+
+  # test algorithm configuration files; list type;
+  algorithms:
```
It is possible to introduce other test objects here rather than algorithm. For a more extendable .yaml, it is suggested to add an abstraction of test object to hold all possible implementations and ensure singleton implementation of the test object.

Commit:
Fix system metric name and it's func mapping

Signed-off-by: JimmyYang20 <yangjin39@huawei.com>

MooreZheng commented on docs/guides/how-to-contribute-leaderboards-or-test-reports.md:
```diff
@@ -17,26 +18,42 @@ the following command:
+- If the test object is ready, you are welcome to contact [Ianvs Owners]. Ianvs owners will connect you and maintainers, in order to receive your test object. Note that when developers submit the test object, developers give maintainers the right to test them. 
 
+For potential maintainers,
+- To maintain the consistence of test environments and test objects, the [leaderboard] submssion is at present calling for acknowledged organizations to apply in charge. Please contact
```
Please contact [Ianvs Owners].

MooreZheng reviewed (commented):
/lgtm

jaypume reviewed (commented):
/lgtm

MooreZheng committed:
Merge pull request #15 from JimmyYang20/v0.1

Fix system metric name and it's func mapping

MooreZheng commented on docs/guides/how-to-contribute-algorithms.md:
```diff
@@ -1 +1,21 @@
+
+
+
+Before start, it should be known that Ianvs testing algorithm development depends on [Sedna Lib]. The following is recommended contribute workflow:
```
Background: 
Ianvs serves as testing tools for test objects, e.g., algorithms. Ianvs does NOT include code directly on test obejct. Alogorithms serve as typical test objects in Ianvs and detailed algorithms are thus NOT included in this Ianvs python file. As for the details of example test obejcts, e.g., algorithms, please refer to third party packages in Ianvs example. For example, AI workflow and interface pls refer to sedna and module implementation pls refer to third party package like FPN_TensorFlow and Sedna IBT algortihm.

MooreZheng replied:
1. release a repo independent of sedna and ianvs, but interface should still follow the sig ai algorithm interface to launch ianvs. 
2. intergrated into sedna so that ianvs can use directly. in this case, you can connect sedna owners for help.

jaypume commented on docs/guides/how-to-install-ianvs.md:
```diff
@@ -0,0 +1,70 @@
+## Install ianvs on Linux
+
+
+### Create virtualenv
```
Add why we need a virtualenv.

jaypume commented on docs/guides/how-to-install-ianvs.md:
````diff
@@ -0,0 +1,70 @@
+
+### Create virtualenv
+```shell
+sudo apt install -y virtualenv
````
1.  It is suggested to change `apt` to `apt-get`, also the following files.
2. This is a guide for Ubuntu, and it should be declared.

jaypume commented on docs/guides/how-to-install-ianvs.md:
````diff
@@ -0,0 +1,70 @@
+source ~/venv/ianvs/bin/activate
+```
+
+> If you prefer conda, you can create a python environment by referring to the [creating steps](https://docs.conda.io/projects/conda/en/latest/user-guide/tasks/manage-environments.html#creating-an-environment-with-commands) provided by conda. 
````
Put this description to the start of section.

jaypume commented on docs/guides/how-to-install-ianvs.md:
```diff
@@ -0,0 +1,70 @@
+sudo apt update
+sudo apt install libgl1-mesa-glx -y
+cd ~/ianvs
+python -m pip install third_party/*
```
Update the command to the latest version.

MooreZheng commented on docs/guides/how-to-install-ianvs.md:
```diff
@@ -0,0 +1,70 @@
+
+
+
+## About Windows
```
Move to the start.

jaypume committed:
fix document reference link

Signed-off-by: Jie Pu <pujie2@huawei.com>

Commit:
Add the field test_object in benchmarkingjob

Signed-off-by: JimmyYang20 <yangjin39@huawei.com>

Commit:
Add comments to emphasize that ianvs is only the test platform

Signed-off-by: JimmyYang20 <yangjin39@huawei.com>

MooreZheng committed:
Merge pull request #16 from jaypume/doc

fix document reference link

MooreZheng committed:
Merge pull request #17 from JimmyYang20/v0.1

Add the field test_object in benchmarkingjob

MooreZheng committed:
Merge pull request #18 from JimmyYang20/v0.1-fixbug

Add comments to emphasize that ianvs is only the test platform

jaypume committed:
Update section: how-to-contribute-algorithms

Signed-off-by: Jie Pu <pujie2@huawei.com>

MooreZheng committed:
Add test object in image and revise index

Signed-off-by: Zimu Zheng <zimu.zheng@hotmail.com>

MooreZheng committed:
Revise procedures of learning paradigms

Signed-off-by: Zimu Zheng <zimu.zheng@hotmail.com>

MooreZheng committed:
Capitalize first character in install doc

Signed-off-by: Zimu Zheng <zimu.zheng@hotmail.com>

Commit:
optimize code of fpn and metric func f1_score

Signed-off-by: JimmyYang20 <yangjin39@huawei.com>

jaypume committed:
Add comments for syphix configuration.

Signed-off-by: Jie Pu <pujie2@huawei.com>

Commit:
Modify third_party's address

Signed-off-by: JimmyYang20 <yangjin39@huawei.com>

jaypume committed:
Update section: how-to-contribute-test-environments

Signed-off-by: Jie Pu <pujie2@huawei.com>

jaypume committed:
Add Ianvs v0.1.0 release note

Signed-off-by: Jie Pu <pujie2@huawei.com>

Commit:
Add ianvs configuration docs

1. how to config benchmarkingjob
2. how to config testenv
3. how to config algorithm

Signed-off-by: JimmyYang20 <yangjin39@huawei.com>

jaypume committed:
Add txt annotation description

Signed-off-by: Jie Pu <pujie2@huawei.com>

jaypume committed:
Add hyperparameter combination description

Signed-off-by: Jie Pu <pujie2@huawei.com>

Commit:
Fix test reports

1. Add effect display of incremental learning
2. Fix configuration info of test reports

Signed-off-by: JimmyYang20 <yangjin39@huawei.com>

Commit:
Update leaderboard

Signed-off-by: JimmyYang20 <yangjin39@huawei.com>

MooreZheng committed:
Merge pull request #21 from MooreZheng/patch-01

Revise procedures of learning paradigms

MooreZheng committed:
Merge pull request #22 from MooreZheng/patch-02

Capitalize first character in install doc

MooreZheng committed:
Merge pull request #31 from jaypume/doc-how-to-config-algorithm

Add hyperparameter combination description

MooreZheng committed:
Merge pull request #30 from jaypume/doc-txt-annotation-desc

Add txt annotation description

MooreZheng committed:
Merge pull request #29 from jaypume/doc-release-note

Add Ianvs v0.1.0 release note

MooreZheng committed:
Merge pull request #28 from jaypume/patch-test-report

Update section: how-to-contribute-test-environments

MooreZheng committed:
Merge pull request #19 from jaypume/doc

Update section: how-to-contribute-algorithms

MooreZheng committed:
Merge pull request #24 from jaypume/patch-01

Add comments for syphix configuration.

MooreZheng committed:
Merge pull request #25 from JimmyYang20/v0.1

Modify third_party's address

MooreZheng committed:
Merge pull request #32 from JimmyYang20/v0.1-report

Fix test reports

MooreZheng committed:
Merge pull request #23 from JimmyYang20/v0.1-fixbug

optimize code of fpn and metric func f1_score

MooreZheng committed:
Merge pull request #33 from JimmyYang20/v0.1-leaderboard

Update leaderboard

Commit:
Fix config info in docs

Signed-off-by: JimmyYang20 <yangjin39@huawei.com>

Commit:
Add ianvs command line usage

Signed-off-by: JimmyYang20 <yangjin39@huawei.com>

Commit:
Merge branch 'v0.1' into v0.1-config

jaypume committed:
Merge pull request #26 from JimmyYang20/v0.1-fix-command

Add ianvs command line usage

Commit:
Merge branch 'v0.1' into v0.1-config

jaypume committed:
Merge pull request #27 from JimmyYang20/v0.1-config

Add ianvs configuration docs

jaypume committed:
Merge pull request #35 from JimmyYang20/v0.1

Fix config info in docs

Commit:
Fix dataset info of pcb-aoi

Signed-off-by: JimmyYang20 <yangjin39@huawei.com>

jaypume committed:
Merge pull request #36 from JimmyYang20/v0.1-data

Fix dataset info of pcb-aoi

MooreZheng committed:
Add owners, gitignore and revise arch

Signed-off-by: Zimu Zheng <zimu.zheng@hotmail.com>

MooreZheng committed:
Merge pull request #37 from MooreZheng/v0.1

Add owners, gitignore and revise arch

jaypume committed:
Update README.md

jaypume committed:
Update README.md
